### PR TITLE
feat(leadspace): updating to v2

### PR DESCRIPTION
**Pull request by IgnacioBecerra**
### Related Ticket(s)
[#10788](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/10788)

### Description
This PR adds the new `highlight` and `type-style` properties to the `c4d-leadspace-heading` component, as well as takes care of some of the spacing updates from the latest design specs.

### Changelog

**New**

- adds the new `highlight` and `type-style` properties for leadspace heading

**Changed**

- updated spacing and story knobs

**Removed**

- centered leadspace stories

### CODE REVIEW

File: packages/styles/scss/components/leadspace/_leadspace.scss [hunk 1]
```diff
@@ -22,8 +20,56 @@
...
+      theme: themes.$g10,
+      value: #0F62FE,
+    ),
+    (
+      theme: themes.$g90,
+      value: #78A9FF,
+    ),
+    (
+      theme: themes.$g100,
+      value: #78A9FF,
```
Comment (ariellalgilmore) [inline packages/styles/scss/components/leadspace/_leadspace.scss:47]:
```suggestion
  fallback: $blue-60,
  values: (
    (
      theme: themes.$white,
      value: $blue-60,
    ),
    (
      theme: themes.$g10,
      value: $blue-60,
    ),
    (
      theme: themes.$g90,
      value: $blue-40,
    ),
    (
      theme: themes.$g100,
      value:$blue-40,
```

File: packages/styles/scss/components/leadspace/_leadspace.scss [hunk 2]
```diff
@@ -473,13 +563,48 @@ $btn-min-width: 26;
...
+  }
+
+  :host(#{$c4d-prefix}-leadspace-heading[type-style='display-01']) {
+    @include type-style(display-01, true);
+  }
+
+  :host(#{$c4d-prefix}-leadspace)[size='super'] {
+    @include breakpoint(lg) {
+      .#{$prefix}--leadspace__overlay {
+        height: 640px;
```
Comment (ariellalgilmore) [inline packages/styles/scss/components/leadspace/_leadspace.scss:587]: should we convert this to rem?
```suggestion
        height: rem(640px);
```

File: packages/styles/scss/components/leadspace/_leadspace.scss [hunk 3]
```diff
@@ -473,13 +563,48 @@ $btn-min-width: 26;
...
+      }
+    }
+  }
+
+  :host(#{$c4d-prefix}-leadspace)[size='tall'] {
+    @include breakpoint(lg) {
+      .#{$prefix}--leadspace__overlay {
+        height: 560px;
```
Comment (ariellalgilmore) [inline packages/styles/scss/components/leadspace/_leadspace.scss:595]:
```suggestion
        height: rem(560px);
```

File: packages/styles/scss/components/leadspace/_leadspace.scss [hunk 4]
```diff
@@ -473,13 +563,48 @@ $btn-min-width: 26;
...
+      }
+    }
+  } 
+  
+  :host(#{$c4d-prefix}-leadspace)[size='medium'],
+  :host(#{$c4d-prefix}-leadspace)[size='short'] {
+
+    @include breakpoint(lg) {
+      .#{$prefix}--leadspace__overlay {
+        height: 480px;
```
Comment (ariellalgilmore) [inline packages/styles/scss/components/leadspace/_leadspace.scss:605]:
```suggestion
        height: rem(480px);
```

File: packages/web-components/.storybook/container.scss [hunk 5]
```diff
@@ -15,10 +15,12 @@
 
 @use '@carbon/ibmdotcom-styles/scss/internal/callout/tokens' as callout-tokens;
 @use '@carbon/styles/scss/components/button/tokens' as button-tokens;
+@use '@carbon/ibmdotcom-styles/scss/components/leadspace' as leadspace;
```
Comment (ariellalgilmore) [inline packages/web-components/.storybook/container.scss:18]: wondering if we can seperate out the tokens like callout out does in separate file, so they only need to import those if they use a different theme. and then also add it to the dotcom [themes cdn file ](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/feat/carbon-for-ibm-dotcom-v2/packages/web-components/src/globals/scss/themes.scss)

File: packages/web-components/src/components/leadspace/leadspace.scss [hunk 6]
```diff
@@ -8,73 +8,6 @@
 @use '@carbon/styles/scss/breakpoint' as *;
 @use '@carbon/styles/scss/config' as *;
 @use '@carbon/styles/scss/spacing' as *;
-@use '@carbon/ibmdotcom-styles/scss/components/leadspace';
+@use '../../../../styles/scss/components/leadspace';
```
Comment (ariellalgilmore) [inline packages/web-components/src/components/leadspace/leadspace.scss:11]:
```suggestion
@use '@carbon/ibmdotcom-styles/scss/components/leadspace';
```

File: packages/web-components/src/components/leadspace/leadspace.scss [hunk 7]
```diff
@@ -8,73 +8,6 @@
...
-    background-position: 0% 50%;
-  }
-  50% {
-    background-position: 100% 50%;
-  }
-  100% {
-    background-position: 0% 50%;
-  }
-}
+@use '@carbon/ibmdotcom-styles/scss/globals/vars' as *;
```
Comment (ariellalgilmore) [inline packages/web-components/src/components/leadspace/leadspace.scss:13]:
```suggestion

```

File: packages/web-components/src/components/leadspace/leadspace.ts [hunk 8]
```diff
@@ -149,6 +149,10 @@ class C4DLeadSpace extends StableSelectorMixin(LitElement) {
         child.slot = 'image';
       }
     });
+
+    if(this.size === 'short') {
+      this.querySelector('c4d-leadspace-heading')?.setAttribute('type-style', 'fluid-heading-05');
```
Comment (ariellalgilmore) [inline packages/web-components/src/components/leadspace/leadspace.ts:154]:
```suggestion
      this.querySelector(`${c4dPrefix}-leadspace-heading`)?.setAttribute('type-style', 'fluid-heading-05');
```

**Review by ariellalgilmore — COMMENTED** [review]
The highlight looks awesome!! just some cleanup comments

File: packages/styles/scss/components/leadspace/_leadspace.scss (hunk 2 above)
Comment (ariellalgilmore) [reply packages/styles/scss/components/leadspace/_leadspace.scss:587]: oh yeah i have seen that! you might be missing the utility import

Comment (ariellalgilmore) [reply packages/styles/scss/components/leadspace/_leadspace.scss:587]: oh it should be this import `@use '@carbon/styles/scss/utilities' as *;`

**Comment by RichKummer** [comment]
Hey @IgnacioBecerra , the highlight and type swapping look great! I noticed a few spacing issues `@sm` and `@md`:

1. The spacing looks off at the smaller breakpoints. There should be `48px` between the heading and body copy. Finally, there should be `32px` spacing between the button group and the image/video if there is one. It looks like the spacing may be stacking when it renders?

Screenshot 2023-09-21 at 5 35 42 PM

2. When `@sm`, the body copy runs off the view instead of wrapping at the edge. It looks like this area is scrollable horizontally, not sure what's happening at this breakpoint 🤔 
Screenshot 2023-09-21 at 5 38 35 PM

3. The `button group` looks very condensed compared to [Olivia's button group ](https://www.figma.com/file/HFWHrSPZWdHd8ilqR5QFAk/Button-group?type=design&node-id=119%3A3358&mode=design&t=7BSxiyCMJRemo0uE-1)updates. There may be a separate PR for this but just want to make sure something in `lead space` isn't effecting it here.
Screenshot 2023-09-21 at 5 35 42 PM

**Comment by IgnacioBecerra** [comment]
@RichKummer Addressed comments!

**Review by ariellalgilmore — APPROVED** [review]
just a few small things, but otherwise LGTM!

**Review by RichKummer — COMMENTED** [review]
@IgnacioBecerra , the vertical spacing and button group updates look great! I noticed a few things about horizontal spacing:

1. For `@sm`, the heading looks like it is being condensed too much. It should extend the full width of the container (ultimately with `16px` on both left and right for `@sm`.
Screenshot 2023-09-26 at 3 23 32 PM

Specs:
Screenshot 2023-09-26 at 3 27 41 PM

2. When `@md`, both the heading and subheading should extend six columns - 32px. It looks like this is a little off (or at the very least they're not equal).
Screenshot 2023-09-26 at 3 29 55 PM

Specs:
Screenshot 2023-09-26 at 3 29 32 PM

3. The height of `Super` is coming up as `672px` when it should be `640px` on desktop sizes. The others are coming in with a greater height than they should as well. Tall shoiuld be 560px, Medium should be 480px, and Short should be 320px. 
Screenshot 2023-09-26 at 3 33 09 PM

4. The subheading should be more condensed width wise at desktop sizes. It is rendering as `#text` which might be off when compared to the heading? See [design specs:](https://www.figma.com/file/Gfb6xdHwKdBgNloNqlSbnG/Lead-space-v2?type=design&node-id=1015%3A13228&mode=design&t=n7frr2iNacUrw1Yk-1)

Screenshot 2023-09-26 at 3 36 42 PM

**Comment by IgnacioBecerra** [comment]
@RichKummer addressed your comments! I might've been checking the inner container for the heights instead of the whole component since it was adding an extra 32px from the top padding haha.

**Comment by RichKummer** [comment]
@IgnacioBecerra this is looking great! The vertical spacing issues look resolved. I did notice that the `Short` size for lead space is coming in at `480px` height when it should render with `320px` height.

**Review by RichKummer — APPROVED** [review]
Thanks for the last minute fix @IgnacioBecerra , this one LGTM 🔥 🔥 🔥

File: packages/styles/scss/components/leadspace/_tokens.scss [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+//
+
+@use '@carbon/colors' as *;
+@use '@carbon/styles/scss/themes';
+@use '@carbon/styles/scss/theme' as *;
+@use '@carbon/styles/scss/utilities/component-tokens';
+
+$btn-min-width: 26;
+
+$leadspace-heading-highlight: (
```
Comment (kennylam) [inline packages/styles/scss/components/leadspace/_tokens.scss:15]: I think all of the highlight color token logic could be removed by simply using `$link-primary`, no? See the [Link](https://carbondesignsystem.com/guidelines/color/tokens) section of the color guidance page.

**Review by kennylam — COMMENTED** [review]
Looks great @IgnacioBecerra! Just a few comments.

File: packages/styles/scss/components/leadspace/_tokens.scss (hunk 9 above)
Comment (IgnacioBecerra) [reply packages/styles/scss/components/leadspace/_tokens.scss:15]: That's true! I didn't realize there was already a token for that color mapping. I'll remove the token file and use this token instead

fyi @RichKummer, the [specs](https://www.figma.com/file/Gfb6xdHwKdBgNloNqlSbnG/Lead-space-v2?type=design&node-id=1021-36034&mode=design&t=lGZeOZtq02WhoAAA-4) can be updated to include the token instead 😄 👍

**Review by kennylam — APPROVED** [review]
LGTM!

**Pull request merged**